include/cantera/kinetics/MultiRateBase.h

```diff
@@ -55,6 +55,36 @@ class MultiRateBase
     //! @param kf  array of rate constants
     virtual void getRateConstants(double* kf) = 0;
 
+    //! Evaluate all rate constant temperature derivatives handled by the evaluator;
+    //! which are multiplied with the array of rate-of-progress variables.
+    //! Depending on the implementation of a rate object, either an exact derivative or
+    //! a numerical approximation may be used.
+    //! @param rop  array of rop, which is modified by the method
+    //! @param kf  array of forward rate constants (numerical derivative only)
+    //! @param deltaT  relative temperature perturbation (numerical derivative only)
+    virtual void processRateConstants_ddT(double* rop,
+                                          const double* kf,
+                                          double deltaT) = 0;
+
+    //! Evaluate all rate constant pressure derivatives handled by the evaluator;
+    //! which are multiplied with the array of rate-of-progress variables.
+    //! @param rop  array of rop, which is modified by the method
+    //! @param kf  array of forward rate constants
+    //! @param deltaP  relative pressure perturbation
+    virtual void processRateConstants_ddP(double* rop,
+                                          const double* kf,
+                                          double deltaP) = 0;
+
+    //! Evaluate all rate constant third-body derivatives handled by the evaluator;
+    //! which are multiplied with the array of rate-of-progress variables.
+    //! @param rop  array of rop, which is modified by the method
+    //! @param kf  array of forward rate constants
+    //! @param deltaM  relative perturbation of third-body concentrations;
+    //!     if deltaM < 0, rop entries not affected by M are not set to zero
+    virtual void processRateConstants_ddM(double* rop,
```

Comment:
I am hesitant doing this, as it really is a different context. Also, note that `GasKinetics` uses uppercase `M` to denote third-bodies elsewhere (also, this method is called by both `(process_)ddX` and `(process_)ddC`).

Reply:
I suppose this is a bit theoretical -- for all current rate parameterizations, if the rate constant depends on the total concentration, it is expressed as a "third body" where the additional notion of efficiencies come in, and I can see wanting to avoid conflating a general dependence on total concentration with concentrations of specific species that comes in though non-unity efficiencies. So, I'm fine with leaving this as is.